nexus/db-queries/src/db/datastore/db_metadata.rs

```diff
@@ -340,6 +662,252 @@ impl DataStore {
+            .values(new_nexus)
+            .on_conflict(dsl::nexus_id)
+            .do_update()
+            .set(dsl::state.eq(excluded(dsl::state)))
+            .execute_async(&*self.pool_connection_unauthorized().await?)
+            .await
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
+
+        Ok(())
+    }
+
+    /// Registers multiple Nexus instances as having active access to the database
+    pub async fn database_nexus_access_insert_multiple(
+        &self,
+        nexus_ids: &[OmicronZoneUuid],
+        state: DbMetadataNexusState,
+    ) -> Result<(), Error> {
+        use nexus_db_schema::schema::db_metadata_nexus::dsl;
+
+        let new_nexuses: Vec<DbMetadataNexus> = nexus_ids
+            .iter()
+            .map(|&nexus_id| DbMetadataNexus::new(nexus_id, state))
+            .collect();
+
+        diesel::insert_into(dsl::db_metadata_nexus)
+            .values(new_nexuses)
+            .on_conflict(dsl::nexus_id)
+            .do_update()
+            .set(dsl::state.eq(excluded(dsl::state)))
+            .execute_async(&*self.pool_connection_unauthorized().await?)
+            .await
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
+
+        Ok(())
+    }
+
+    /// Initializes Nexus database access records from a blueprint using an
+    /// existing connection
+    ///
+    /// This function finds all Nexus zones in the given blueprint and creates
+    /// active database access records for them. Used during RSS rack setup.
+    ///
+    /// Returns an error if:
+    /// - Any db_metadata_nexus records already exist (should only be called
+    /// during initial setup)
+    pub async fn initialize_nexus_access_from_blueprint_on_connection(
+        &self,
+        conn: &async_bb8_diesel::Connection<DbConnection>,
+        blueprint_id: Uuid,
+    ) -> Result<(), Error> {
+        use nexus_db_model::ZoneType;
+        use nexus_db_schema::schema::bp_omicron_zone::dsl as zone_dsl;
+        use nexus_db_schema::schema::db_metadata_nexus::dsl;
+
+        // Ensure no db_metadata_nexus records already exist
+        let any_records_exist =
+            Self::database_nexus_access_any_exist_on_connection(conn).await?;
+        if any_records_exist {
+            return Err(Error::internal_error(
+                "Cannot initialize Nexus access from blueprint: db_metadata_nexus records already exist. \
+                This function should only be called during initial rack setup.",
+            ));
+        }
+
+        // Query all Nexus zones from the blueprint
+        let nexus_zone_ids: Vec<Uuid> = zone_dsl::bp_omicron_zone
+            .filter(zone_dsl::blueprint_id.eq(blueprint_id))
+            .filter(zone_dsl::zone_type.eq(ZoneType::Nexus))
+            .select(zone_dsl::id)
+            .load_async(conn)
+            .await
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
```

Comment:
I believe the sole caller of this function already has the blueprint handy so I think we could replace this with functions that access that blueprint.

If for some reason we want to keep this, I think we need to be filtering on the zone disposition and maybe the nexus_generation.  Though it would seem pretty unlikely to have expunged zones here or any from an older generation.